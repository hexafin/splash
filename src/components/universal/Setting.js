import React from "react";
import {
    StyleSheet,
    View,
    Text,
    Switch,
    TouchableOpacity,
} from "react-native"
import {colors} from "../../lib/colors"

const Setting = ({toggleCallback, toggleState, infoCallback, help}) => {
	return (
	    <View style={styles.container}>
	    	<View style={{justifyContent: 'space-between', flexDirection: 'row'}}>
		    	<View style={styles.topRow}>
			    	<Text style={styles.title}>Lock your account</Text>
			    	{help && <TouchableOpacity style={styles.helpCircle} onPress={infoCallback}>
			    		<Text style={styles.questionMark}>?</Text>
			    	</TouchableOpacity>}
		    	</View>
		    	<Switch onTintColor={'#5153E9'} onValueChange={toggleCallback} value={toggleState}/>
	    	</View>
	    	<Text style={styles.description}>Set a four digit passcode to secure your Splash wallet.</Text>
	    </View>
	)
}

const styles = StyleSheet.create({
	container: {
		paddingVertical: 5,
		backgroundColor: colors.white,
	},
	topRow: {
		flexDirection: 'row',
		alignItems: 'center',
		paddingBottom: 5,
	},
	title: {
		fontSize: 15,
		fontWeight: "500",
		paddingRight: 10,
		color: colors.nearBlack,
	},
	helpCircle: {
		width: 23,
		height: 23,
		backgroundColor: '#EFEFFD',
		borderRadius: 11.5,
		justifyContent: 'center',
		alignItems: 'center'
	},
	questionMark: {
		color: colors.purple,
		fontSize: 15,
	},
	description: {
		fontSize: 14,
		color: '#919191',
		paddingRight: 78,
	},
})

export default Setting
