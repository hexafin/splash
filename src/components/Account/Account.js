import React, { Component } from "react";
import {
	View,
	Alert,
	Text,
	StyleSheet,
	Image,
	TouchableOpacity,
	Dimensions
} from "react-native";
import { colors } from "../../lib/colors";
import { defaults, icons } from "../../lib/styles";
import { isIphoneX } from "react-native-iphone-x-helper"
import FlatBackButton from "../universal/FlatBackButton"
import {Input} from "../universal/Input"
import Setting from "../universal/Setting"
import api from '../../api'

const Account = ({splashtag, userId, logout, navigation, toggleLockout, lockoutEnabled}) => {

		const handleLogout = () => {
			Alert.alert(
			  'Confirm Delete',
			  'Are you sure you want to delete your account? Your funds will be irretrievable.',
			  [
			    {text: 'No', onPress: () => console.log('Cancel Pressed'), style: 'cancel'},
			    {text: 'Yes', onPress: () => logout(userId)},
			  ],
			)
		}

		const handleLockoutSwitch = (enabled) => {
			console.log(enabled)
			if (enabled) {
				navigation.navigate('SetPasscode', {
					successCallback: (newCode) => {

						api.AddToKeychain(userId, 'passcode', newCode).then(() => {
							navigation.navigate('SwipeApp')	
							toggleLockout(enabled)
						}).catch(e => console.log(e))

					}
				})
			} else {
				navigation.navigate("Unlock", {
					successCallback: () => {

						api.AddToKeychain(userId, 'passcode', '').then(() => {
							console.log('made it', toggleLockout, enabled)
							navigation.navigate('SwipeApp')	
							toggleLockout(enabled)
						}).catch(e => console.log(e))

					}
				})
			}
		}

		return (
			<View style={styles.container}>
				<View style={styles.header}>
					<Text style={styles.title}>@{splashtag}</Text>
				</View>
				<View style={styles.body}>
					<View style={styles.section}>
						<Text style={styles.sectionText}>Your account</Text>
						<Text style={styles.splashtagText}>Splashtag</Text>
						<TouchableOpacity onPress={() => navigation.navigate("UpdateUsername")}>
							<View pointerEvents='none'>
								<Input editable={false} input={{value: splashtag}} />
							</View>
						</TouchableOpacity>
					</View>
					<View style={styles.section}>
						<Text style={styles.sectionText}>Settings</Text>
						<Setting toggleCallback={handleLockoutSwitch} toggleState={lockoutEnabled} help={true}/>
						<TouchableOpacity style={{marginTop: 42}} onPress={handleLogout}>
							<Text style={styles.logoutText}>Delete Account</Text>
						</TouchableOpacity>
					</View>
				</View>
			</View>
		);
}

const styles = StyleSheet.create({
	container: {
		...defaults.container,
		backgroundColor: "transparent"
	},
	header: {
		height: 150,
		justifyContent: "center",
		alignItems: "center",
		marginBottom: 40,
		paddingTop: 60
	},
	title: {
		color: colors.white,
		fontSize: 27,
		fontWeight: "700"
	},
	section: {
		paddingBottom: 40
	},
	body: {
		flex: 1,
		padding: 24,
		flexDirection: "column",
		backgroundColor: colors.white
	},
	accountText: {
		fontSize: 18,
		fontWeight: "500",
		color: colors.nearBlack
	},
	splashtagText: {
		fontSize: 16,
		fontWeight: "700",
		color: "#B3B3B3",
		paddingVertical: 6,
	},
	sectionText: {
		fontSize: 18,
		fontWeight: '600',
		paddingBottom: 10,
		color: colors.nearBlack, 
	},
	logoutText: {
		fontSize: 17,
		fontWeight: "700",
		color: '#3F41FA',
		alignSelf: 'center',
		backgroundColor: 'rgba(0,0,0,0)'
	}
});

export default Account;