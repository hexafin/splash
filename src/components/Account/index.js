import Account from "./Account"
import { connect } from "react-redux"
import { bindActionCreators } from "redux"
import {resetUser, toggleLockout, setBiometric} from "../../redux/user/actions"
import { ToggleNetwork } from '../../redux/crypto/actions'
import {resetTransactions} from "../../redux/transactions/actions"
import {resetOnboarding} from "../../redux/onboarding/actions"
import {resetCrypto} from "../../redux/crypto/actions"
import NavigatorService from "../../redux/navigator"
import api from '../../api'


const showLockInfo = () => {
  return {
    type: 'SHOW_MODAL',
    modalType: 'INFO',
    modalProps: {
    	title: 'Set a Passcode',
    	body: 'Set a four-digit passcode to secure your Splash wallet. After closing the app for 5 minutes you will be asked to authenticate using the passcode or biometrics.',
    },   
  }
}

const mapStateToProps = state => {
	let network = 'testnet'
	if (typeof state.crypto.wallets.BTC !== 'undefined') {
		network = state.crypto.wallets.BTC.network
	}
	return {
		splashtag: state.user.entity.splashtag,
		userId: state.user.id,
		lockoutEnabled: state.user.lockoutEnabled,
		bitcoinNetwork: network,
		biometricEnabled: state.user.biometric,
	}
}

const mapDispatchToProps = dispatch => {
    return bindActionCreators({
		logout: (userId) => {
			return (dispatch, getState) => {
				dispatch(resetUser())
				dispatch(resetTransactions())
				dispatch(resetOnboarding())
				dispatch(resetCrypto())
				NavigatorService.navigate("Landing")
				api.DeleteAccount(userId)
			}		
		},
    	toggleLockout,
    	showLockInfo,
    	resetTransactions,
    	ToggleNetwork,
    	setBiometric,
    }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Account)
