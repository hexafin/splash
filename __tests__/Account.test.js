import "react-native";
import React from "react";
import Account from "../src/components/Account/Account";
import Setting from "../src/components/Account/Setting";

it("Account renders correctly", () => {
	const snap = shallow(<Account />);
	expect(snap).toMatchSnapshot();
});

it("Setting renders correctly", () => {
	const snap = shallow(<Setting />);
	expect(snap).toMatchSnapshot();
});
