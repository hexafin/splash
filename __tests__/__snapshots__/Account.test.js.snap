// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account renders correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
      "paddingBottom": 20,
      "paddingTop": 20,
    }
  }
>
  <ScrollView>
    <Component
      style={
        Object {
          "alignItems": "center",
          "height": 150,
          "justifyContent": "center",
          "marginBottom": 40,
          "paddingTop": 60,
        }
      }
    >
      <Component
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 27,
            "fontWeight": "700",
          }
        }
      >
        @
      </Component>
    </Component>
    <Component
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "column",
          "padding": 20,
        }
      }
    >
      <Component
        style={
          Object {
            "paddingBottom": 30,
          }
        }
      >
        <Component
          style={
            Object {
              "color": "#292342",
              "fontSize": 20,
              "fontWeight": "700",
            }
          }
        >
          Your account
        </Component>
        <Component
          style={
            Object {
              "color": "#B3B3B3",
              "fontSize": 16,
              "fontWeight": "700",
              "paddingVertical": 10,
            }
          }
        >
          Splashtag
        </Component>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Component
            pointerEvents="none"
          >
            <Input
              editable={false}
              input={
                Object {
                  "value": undefined,
                }
              }
            />
          </Component>
        </TouchableOpacity>
      </Component>
      <Component
        style={
          Object {
            "paddingBottom": 30,
          }
        }
      >
        <Component
          style={
            Object {
              "color": "#333333",
              "fontSize": 18,
              "fontWeight": "600",
              "paddingBottom": 10,
            }
          }
        >
          Settings
        </Component>
        <Setting
          description="On the mainnet your coins hold real value. If unselected your app will use testnet tokens."
          title="Use undefined mainnet"
          toggleCallback={[Function]}
          toggleState={false}
        />
        <Setting
          description="Set a four digit passcode to secure your Splash wallet."
          help={true}
          title="Set a PIN"
          toggleCallback={[Function]}
        />
        <Setting
          description="Secure your account with Touch ID or Face ID."
          title="Use biometric security"
          toggleCallback={[Function]}
        />
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "marginTop": 42,
            }
          }
        >
          <Component
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "rgba(0,0,0,0)",
                "color": "red",
                "fontSize": 17,
                "fontWeight": "700",
              }
            }
          >
            Delete Account
          </Component>
        </TouchableOpacity>
      </Component>
    </Component>
  </ScrollView>
</Component>
`;

exports[`Setting renders correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "paddingVertical": 5,
    }
  }
>
  <Component
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingBottom": 5,
        }
      }
    >
      <Component
        style={
          Object {
            "color": "#333333",
            "fontSize": 15,
            "fontWeight": "500",
            "paddingRight": 10,
          }
        }
      />
    </Component>
    <Switch
      trackColor="#5153E9"
    />
  </Component>
  <Component
    style={
      Object {
        "color": "#919191",
        "fontSize": 14,
        "paddingRight": 78,
      }
    }
  />
</Component>
`;
